crates/fluvio/src/producer/accumulator.rs

```diff
@@ -25,30 +27,45 @@ use crate::error::Result;
 
 use super::event::EventHandler;
 
-pub(crate) type BatchHandler = (Arc<BatchEvents>, Arc<Mutex<VecDeque<ProducerBatch>>>);
+const RECORD_ENQUEUE_TIMEOUT: Duration = Duration::from_secs(30);
+
+pub(crate) type BatchHandler = (
+    Arc<BatchEvents>,
+    Arc<(Mutex<VecDeque<ProducerBatch>>, Condvar)>,
```

Comment:
Probably should abstract out: 
```
Arc<(Mutex<VecDeque<ProducerBatch>>, Condvar)>
```

as `NewType` so can independently unit test and easy to reason about

Reply:
new internal type created for that